node/src/command.rs

```diff
@@ -273,53 +276,83 @@ pub fn run() -> Result<()> {
-                    crate::service::start_dev_node(
-                        config,
-                        provider_options,
-                        indexer_options,
-                        hwbench,
-                        id,
-                        cli.run.sealing,
-                    )
-                    .await
-                    .map_err(Into::into)
-                } else {
-			        let collator_options = cli.run.collator_options();
-                    let polkadot_cli = RelayChainCli::new(
-                        &config,
-                        [RelayChainCli::executable_name()].iter().chain(cli.relay_chain_args.iter()),
-                    );
-
-                    let parachain_account =
-                        AccountIdConversion::<polkadot_primitives::AccountId>::into_account_truncating(
-                            &id,
-                        );
-
-				    let tokio_handle = config.tokio_handle.clone();
-
-                    let polkadot_config =
-                        SubstrateCli::create_configuration(&polkadot_cli, &polkadot_cli, tokio_handle)
-                            .map_err(|err| format!("Relay chain argument error: {}", err))?;
-
-                    info!("Parachain Account: {parachain_account}");
-
-                    crate::service::start_parachain_node(
-                        config,
-                        polkadot_config,
-                        collator_options,
-                        provider_options,
-                        indexer_options,
-                        id,
-                        hwbench,
-                    )
-                    .await
-                    .map(|r| r.0)
-                    .map_err(Into::into)
-                }
+				match config.network.network_backend {
+					sc_network::config::NetworkBackendType::Libp2p => {
+						if dev_service {
+							crate::service::start_dev_node::<sc_network::NetworkWorker<_, _>>(
+								config,
+								provider_options,
+								indexer_options,
+								hwbench,
+								id,
+								cli.run.sealing,
+							)
+							.await
+							.map_err(Into::into)
+						} else {
+							let collator_options = cli.run.collator_options();
+							let polkadot_cli = RelayChainCli::new(
+								&config,
+								[RelayChainCli::executable_name()].iter().chain(cli.relay_chain_args.iter()),
+							);
+							let tokio_handle = config.tokio_handle.clone();
+							let polkadot_config =
+								SubstrateCli::create_configuration(&polkadot_cli, &polkadot_cli, tokio_handle)
+									.map_err(|err| format!("Relay chain argument error: {}", err))?;
+							crate::service::start_parachain_node::<sc_network::NetworkWorker<_, _>>(
+								config,
+								polkadot_config,
+								collator_options,
+								provider_options,
+								indexer_options,
+								id,
+								hwbench,
+							)
+							.await
+							.map(|r| r.0)
+							.map_err(Into::into)
+						}
+					},
+					sc_network::config::NetworkBackendType::Litep2p => {
```

Comment:
What is the reason behind supporting both network backends?

Reply:
Mostly because `litep2p` is not complete and might cause issues with the file transfers and everything we have to do appart from consensus and gossiping

Reply:
Does the template node support both too?

Reply:
Yes